README.md

```diff
@@ -8,6 +8,38 @@ To install, copy `IntelliJIdea14/Airlift.xml` into `~/Library/Preferences/Intell
 
 To install, copy `IntelliJIdea14/Airlift.xml` into `$HOME/.IdeaIC14/config/codestyles`
 
+## Programming guidelines
+
+* Commit messages should follow the format described [here](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html). Ideally, you should be able to look at the commit and:
+    * be able to tell whether this fix applies to something they may be seeing,
+    * understand the reason why it's broken,
+    * have some context when looking at the actual changes in the commit (trying to reverse-engineer the observed behavior from the code changes can be very hard).
+
+  Commit message are the way developers communicate with their future selves and other
+  colleagues so special care should be taken when writing them.
+* End every file with a newline. To enable this automatically on save in IntelliJ 14: File->Settings->General->Other (section on the right when you click on General)->Check "Ensure line feed at file end on Save".
+* Don't use `else`, if the `if` has a return in it.
+* Don't mix concurrency and inheritance. Concurrent classes should not compose/inherit because that forces you to reason about the memory visibility of each member and understand the locking semantics of the class' parent(s). Generally, inheritance makes concurrency semantics implicit and harder to follow.
+* Favor early termination (return) on certain input values to methods. This improves readability because it divides the method into two sections: the first where you enumerate exit conditions and the second where the actual method logic lives.
+* Inline variables that are used only once. However, if the expression is very complicated and assigning it a descriptive variable name improves readability then that's what you should do.
+* Avoid using Mockito style mocking frameworks in tests because they make tests brittle and less readable. Use such frameworks as a last resort when you're testing against libraries that don't have good abstractions.
+
+## General Java style guidelines
+
+* Do not abbreviate Java variable names: `command` and not `cmd`.
+* Google's Guava is seen as an extension of the JDK. You should use it when
+    * certain functionality doesn't exist in the JDK or
+    * Guava presents you with a clearly superior API and doesn't degrade performance.
+
+  For example, do not use `Guava.checkNotNull()` because the same functionality exists
+  in the JDK as `Objects.requireNonNull()`.
```

Comment:
it is ironic that this is the most quoted example while JDK checkNotNull is not equivalent to checkNotNull (it does not support the parameter templating). Using `requireNonNull(String.format(...))` next to the other Guava checks (`checkState` and `checkArgument`) makes formatting of error messages uneven and counterintuitive. In fact, the Guava developers state (http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Preconditions.html):

--- cut ---
java.util.Objects.requireNonNull()

Projects which use com.google.common should generally avoid the use of Objects.requireNonNull(Object). Instead, use whichever of checkNotNull(Object) or Verify.verifyNotNull(Object) is appropriate to the situation. (The same goes for the message-accepting overloads.)
--- cut ---

so this recommendation also goes against the recommendations in the "Guava explained" documentation: https://code.google.com/p/guava-libraries/wiki/PreconditionsExplained 

I would suggest to be pragmatic here, no dogmatic and follow the Guava documentation (with this one cited example being the obvious exception).

And yes, the modernizer plugin got that one wrong, too.